joblib/hashing.py

```diff
@@ -131,18 +131,23 @@ def save_global(self, obj, name=None, pack=struct.pack):
     # function
     dispatch[type(pickle.dump)] = save_global
 
-    def _batch_setitems(self, items):
+    def _batch_setitems(self, items, obj=None):
+        # Python 3.14 _batch_setitems has an additional obj argument, see
+        # https://github.com/python/cpython/pull/122214
+        additional_args = [] if obj is None else [obj]
         # forces order of keys in dict to ensure consistent hash.
```

Comment:
As this is a pass-through argument where we only want to make sure that the first argument `items` is sorted, I don't really see the point of making this explicit. This increase the burden of maintaining this for each version of python we support, which would overall increase the maintenance burden.
```suggestion
    def _batch_setitems(self, items, obj=None):
        # forces order of keys in dict to ensure consistent hash.

        # Python 3.14 _batch_setitems has an additional obj argument, see
        # https://github.com/python/cpython/pull/122214
        additional_args = [] if obj is None else [obj]
```